src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animalToBite) {
+    if (animalToBite instanceof Herbivore && !animalToBite.hidden) {
+      animalToBite.health -= 50;
+
+      if (animalToBite.health === 0) {
```

Comment:
Better to check if `health` is less or equal to 0
We may change animal health or carnivore damage and it shouldn't require any refactoring